ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -258,22 +251,12 @@ export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
     }
   }
 
-  const legendWidth = legend && legend.position === 'Right' ? 200 : adjustedContentDimensions.width;
-
-  // TODO: account for number of time series returned when adjusting legend spacing
-  let legendHeight = LEGEND_HEIGHT_SM;
-  if (legend && legend.position === 'Right') {
-    legendHeight = contentDimensions?.height || adjustedContentDimensions.height;
-  } else if (adjustedContentDimensions.height >= PANEL_HEIGHT_LG_BREAKPOINT) {
-    legendHeight = LEGEND_HEIGHT_LG;
-  }
-
   // override default spacing, see: https://echarts.apache.org/en/option.html#grid
   const gridLeft = y_axis && y_axis.label ? 30 : 20;
   const gridOverrides: GridComponentOption = {
     left: !echartsYAxis.show ? 0 : gridLeft,
-    right: legend && legend.position === 'Right' ? legendWidth : 20,
-    bottom: legend && legend.position === 'Bottom' ? legendHeight : 0,
+    right: 20,
```

Comment:
So ECharts grid doesn’t need to be adjusted like it was previously? I think I did it that way so that in panel edit drawer when you click to change between Bottom and Right it doesn’t cause a lag / weird jump while the padding was adjusting (might not be a problem anymore with all the other recent changes)

Reply:
Lemme give that use case a double-check to make sure it looks ok with the new layout.

Reply:
Here's a video of what it looks like in the next WIP PR where I actually add the options.

 Is the little period where there's weird overlap with the list legend what you're talking about? I think you don't see it with the table legend because it doesn't have a transparent background, which is something I could fix about the list legend.

https://github.com/perses/perses/assets/396962/ce57a9a5-032c-4a2b-9839-cc71057a92a4

Reply:
Putting `overflow: hidden` on the content wrapper in `ContentWithLegend` seems to help this a bunch. I pushed up a change with that and story for the component that exercises it. How do you feel about this?


https://github.com/perses/perses/assets/396962/ea9bd6ce-ed4d-4988-a7d7-371b7edeb502

Reply:
Digging a little mroe, I think the debouncing in the resize in `EChart` is what causes that delay. I'll take a closer look in the morning to see if there's a clever way to disable the debounce timeout for these one-time changes like a position swap while keeping it for cases like dragging to resize where you definitely want to debounce.

Reply:
ooh interesting, we should talk about that EChart wrapper more sometime! I think long term we're going to need a hybrid of the approach tracing used internally in their wrapper and what we've built so far

And that video looks great 🙌

Reply:
I was nerd sniped and poked at this some more after dinner. There's a bunch of issues with the debouncing logic in `EChart.tsx` that I don't think is actually doing what we want it to. I want to circle back to fixing it, but NOT in this PR, since it's a complex change that will benefit from more thorough, separate review. 

In the short-term, what I did change is making the `useEffect` debounce run on `leading` (i.e. it doesn't wait to run the first time) instead of the default `trailing` (it would wait 200ms for the first run, causing that lag you saw). I also left some comments in the related code annotating what I think is wrong with it. In the morning, I'll make a more detailed ticket around this that I'll try to get to after I get this initial round of MVP table legend PRs up. 

Does that sound ok to you?

Reply:
Also, here's a video showing what it looks like after that switch to `leading. Note that the toggle is nice and snappy now. I also show a resize of a panel, which is theoretically where we want the debouncing to work and it seems the same as before, so I'm not harming anything with the change.


https://github.com/perses/perses/assets/396962/0cce61aa-d8fc-41dd-889b-5e7a80621337